src/sentry/utils/performance_issues/base.py

```diff
@@ -130,11 +131,17 @@ def is_creation_allowed_for_system(self) -> bool:
             return False
 
         try:
-            rate = options.get(system_option)
-        except options.UnknownOption:
-            rate = 0
+            creation_option_value: bool | float | None = options.get(system_option)
+            if isinstance(creation_option_value, bool):
+                return creation_option_value
```

Comment:
I did use `options.lookup_key(key_name).type` as I couldn't figure out how to do the comparison
I tried 
```
type = options.lookup_key(key_name).type
type is bool
```
and
```
type = options.lookup_key(key_name).type
type == bool
```